### PR TITLE
docs: third-party state management and migration guide

### LINKED ISSUE #19696: Docs feedback - Setup store
Hello,

Just a feedback on the documentation from a beginner at Nuxt3 (with a little bit of experience on Nuxt2).
I find the doc to be too abstract on the subject of the store, and its initialization.

With Nuxt2, I used to create a "store" folder and place my JS files inside it.

What about Nuxt3 ?

- Should I use [this way](https://nuxt.com/docs/getting-started/state-management#shared-state) of centralized state management in Nuxt 3, instead of VueX or Pinia ?
- If I need to use Pinia, should I create a "store" folder and place my JS files inside it, like for Nuxt 2 ? It seems the "store" folder [disappeared](https://nuxt.com/docs/guide/directory-structure/server) in the folder structure of Nuxt 3.

Thanks for your help

### PR DESCRIPTION
**Pull request by adrienZ**
resolves #19696

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Added example to do mimic vuex global state management from Nuxt 2 with pinia and nuxt 3.
- Referenced popular modules related to state management in docs

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.getting-started/7.state-management.md [hunk 1]
```diff
@@ -1,10 +1,24 @@
 ---
 navigation.icon: uil:database
-description: Nuxt provides useState composable to create a reactive and SSR-friendly shared state.
+description: Nuxt powerful state management libraries and useState composable to create a reactive and SSR-friendly shared state.
```
Comment (AloisSeckar) [inline docs/1.getting-started/7.state-management.md:3]: The sentence now lacks a verb. Probably `provides` should stay insteand of being replaced? Or did you mean something else?

Comment (adrienZ) [reply docs/1.getting-started/7.state-management.md]: you are right, I should have re-read my self

File: docs/1.getting-started/7.state-management.md [hunk 2]
```diff
@@ -1,10 +1,24 @@
 ---
 navigation.icon: uil:database
-description: Nuxt provides useState composable to create a reactive and SSR-friendly shared state.
+description: Nuxt provides powerful state management libraries and useState composable to create a reactive and SSR-friendly shared state.
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:3]:
```suggestion
description: Nuxt provides powerful state management libraries and the useState composable to create a reactive and SSR-friendly shared state.
```

File: docs/1.getting-started/7.state-management.md [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
 ---
 
 # State Management
 
+Nuxt **used to rely** on the Vuex library to provide global state management, if you are migrating from nuxt 2, please head to [migration guide](/docs/7.migration/2.configuration.md#vuex).
+
+Nuxt is now unopiniated about this topic, choose the right solution for your needs is up to you.
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:10]:
```suggestion
Nuxt is not opiniated about state management, so feel free to choose the right solution for your needs.
```

File: docs/1.getting-started/7.state-management.md [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+However here is some keys to help you make an informed decision.
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:11]:
```suggestion

However here are some pointers to help you make an informed decision.
```

File: docs/1.getting-started/7.state-management.md [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
+
+## Using a third-party library
+
+One of the features Nuxt can give you is a strong modules ecosystem. You will find multiple integrations with the most popular state management library such as:
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:15]:
```suggestion
One of the strengths of Nuxt is its strong modules ecosystem. There are multiple integrations with the most popular state management libraries, including:

```

File: docs/1.getting-started/7.state-management.md [hunk 6]
```diff
@@ -1,10 +1,24 @@
...
+- [Pinia](https://nuxt.com/modules/pinia) the official Vue recommendation.
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:16]:
```suggestion
- [Pinia](https://nuxt.com/modules/pinia) - the official Vue recommendation
```

File: docs/1.getting-started/7.state-management.md [hunk 7]
```diff
@@ -1,10 +1,24 @@
...
+- [Harlem](https://nuxt.com/modules/harlem) immutable global state management.
+- [XState](https://nuxt.com/modules/xstate) approches state as machine and provides visual tools.
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:18]:
```suggestion
- [Harlem](https://nuxt.com/modules/harlem) - immutable global state management
- [XState](https://nuxt.com/modules/xstate) - state machine approach with tools for visualising and testing your state logic
```

File: docs/1.getting-started/7.state-management.md [hunk 8]
```diff
@@ -1,10 +1,24 @@
...
+
+## Using useState
+
 Nuxt provides useState composable to create a reactive and SSR-friendly shared state across components.
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:22]:
```suggestion
Nuxt provides the `useState` composable to create a reactive and SSR-friendly shared state across components.
```

File: docs/1.getting-started/7.state-management.md [hunk 9]
```diff
@@ -1,10 +1,24 @@
...
+However here is some keys to help you make an informed decision.
+
+## Using a third-party library
+
+One of the features Nuxt can give you is a strong modules ecosystem. You will find multiple integrations with the most popular state management library such as:
+- [Pinia](https://nuxt.com/modules/pinia) the official Vue recommendation.
+- [Harlem](https://nuxt.com/modules/harlem) immutable global state management.
+- [XState](https://nuxt.com/modules/xstate) approches state as machine and provides visual tools.
+
+## Using useState
```
Comment (danielroe) [inline docs/1.getting-started/7.state-management.md:20]:
```suggestion
## Using `useState`
```

File: docs/7.migration/2.configuration.md [hunk 10]
```diff
@@ -135,4 +135,54 @@ It is not currently possible to use the Vue 3 migration build with Nuxt 3 RC.
...
+  actions: {
+    increment() {
+      // `this` is the store instance
+      this.counter++
+    },
+  },
+})
+```
+
+Create a [plugin](/docs/2.guide/2.directory-structure/1.plugins.md) file to globalize your store:
```
Comment (danielroe) [inline docs/7.migration/2.configuration.md:164]:
```suggestion
Create a [plugin](/docs/guide/directory-structure/plugins) file to globalize your store:
```

File: docs/7.migration/2.configuration.md [hunk 11]
```diff
@@ -135,4 +135,54 @@ It is not currently possible to use the Vue 3 migration build with Nuxt 3 RC.
...
+  }
+})
+```
+
 If you want to keep using Vuex, you can manually migrate to Vuex 4 following [these steps](https://vuex.vuejs.org/guide/migrating-to-4-0-from-3-x.html).
+
+Once it's done you will need to add the following plugin to your Nuxt app:
+
+```ts [plugins/vuex.ts]
+import store from '@/store'
```
Comment (danielroe) [inline docs/7.migration/2.configuration.md:183]:
```suggestion
import store from '~/store'
```

File: docs/7.migration/2.configuration.md [hunk 12]
```diff
@@ -135,4 +135,54 @@ It is not currently possible to use the Vue 3 migration build with Nuxt 3 RC.
...
+      this.counter++
+    },
+  },
+})
+```
+
+Create a [plugin](/docs/2.guide/2.directory-structure/1.plugins.md) file to globalize your store:
+
+```ts [plugins/pinia.ts]
+import { useMainStore } from '@/store'
```
Comment (danielroe) [inline docs/7.migration/2.configuration.md:167]:
```suggestion
import { useMainStore } from '~/store'
```

**Review by danielroe — APPROVED** [review]
This is great! A few language tweaks, and I think I'll move this section to the end of the page after the `useState` documentation. But otherwise 👌

**Comment by danielroe** [comment]
Let me know what you think. This is good to go from my point of view - and thank you so much for this! ❤️

**Comment by titouandk** [comment]
Thank you all !

**Comment by titouandk** [comment]
Isn't the "stores" folder plural tho?

**Comment by adrienZ** [comment]
> Isn't the "stores" folder plural tho?

Note in nuxt 2 https://nuxtjs.org/docs/directory-structure/store

**Pull request merged**